src/platform/x86/x86_64/loader.S

```diff
@@ -0,0 +1,216 @@
+#include <chal/shared/cos_config.h>
+#include "chal_asm_inc.h"
+
+.set MULTIBOOT_PAGE_ALIGN,  1<<0
+.set MULTIBOOT_MEMINFO,     1<<1
+.set MULTIBOOT_AOUT_KLUDGE, 1<<16
+.set MULTIBOOT_MAGIC,       0x1BADB002
+.set MULTIBOOT_FLAGS,       MULTIBOOT_PAGE_ALIGN | MULTIBOOT_MEMINFO
+.set MULTIBOOT_CHECKSUM,    -(MULTIBOOT_MAGIC + MULTIBOOT_FLAGS)
+
+.section .multiboot
+.align 4
+.long MULTIBOOT_MAGIC
+.long MULTIBOOT_FLAGS
+.long MULTIBOOT_CHECKSUM
+
+.set STACKSIZE, 0x1000
+.comm stack, (STACKSIZE * NUM_CPU), 0x1000
+
+.globl loader
+loader:
```

Comment:
There are a lot of fixmes here. This seems like it needs to be cleaned up.

I'm a little surprized with the use of movq before switching to 64 bit, but I don't remember any of the details here, so it is likely OK.